frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -184,7 +183,12 @@ private void downloadLicenseInfo(ResourceRequest request, ResourceResponse respo
                 List<AttachmentUsage> attachmentUsages = ProjectPortletUtils.makeAttachmentUsages(project, selectedReleaseAndAttachmentIds,
                         excludedLicensesPerAttachmentId);
                 AttachmentService.Iface attachmentClient = thriftClients.makeAttachmentClient();
-                attachmentClient.replaceAttachmentUsages(Source.projectId(project.getId()), attachmentUsages);
+                if (attachmentUsages.isEmpty()) {
+                    attachmentClient.deleteAttachmentUsagesByUsageDataType(Source.projectId(project.getId()),
```

Comment:
Why does the `downloadLicenseInfo` function delete attachmentUsages? This function is way to long and to complicated. A refactoring might be good here.

Reply:
I've extracted the part that deals with replacing stored attachment usages